src/meta/processors/admin/RestoreProcessor.cpp

```diff
@@ -171,7 +238,13 @@ void RestoreProcessor::process(const cpp2::RestoreMetaReq& req) {
         return;
       }
 
-      // TODO(spw): need to replace the registered machine
+      result = replaceHostInMachine(h.get_from_host(), h.get_to_host(), true);
```

Comment:
Since there are at least three function need to write data, maybe you could merge all data into one batch, so as to make sure all data written is all or nothing. You could see other meta processor for reference (only write a `WriteBatch`), in that way, `multiPutWithoutReplicator` and `multiRemoveWithoutReplicator` could be unified as well.

Reply:
done~